lib/node/node.js

```diff
@@ -468,3 +468,56 @@ function validate(validator, params, expectedParamName) {
 
     return param;
 }
+
+Node.prototype.computeRequirements = function(databaseService, limits, callback) {
+    // By default simply compute maximum of the inputs' number of rows.
+    // TODO: if the most common multi-input analysis is some kind of join we should use
+    // the product of the input numberOfRows instead
+    var maxRows = Math.max.apply(
+        null,
+        this.inputNodes.map(function(node) { return node.estimatedRequirements.numberOfRows || 0; })
+    );
+    if (maxRows < 0) {
+        maxRows = 0;
+    }
+    this.estimatedRequirements = {
+        numberOfRows: maxRows
+    };
+    this.limits = {
+        maximumNumberOfRows: getNodeLimit(limits, this.getType(), 'maximumNumberOfRows', 1000000)
```

Comment:
Since limit values should be definable in Redis, maybe we should use snake_case instead of camelCase?